oci/archive/oci_dest.go

```diff
@@ -0,0 +1,188 @@
+package archive
+
+import (
+	"io"
+	"os"
+
+	"github.com/containers/image/types"
+	"github.com/containers/storage/pkg/archive"
+	"github.com/pkg/errors"
+)
+
+type ociArchiveImageDestination struct {
+	ref          ociArchiveReference
+	unpackedDest types.ImageDestination
+	tempDir      string
+}
+
+// newImageDestination returns an ImageDestination for writing to an existing directory.
+func newImageDestination(ref ociArchiveReference, ctx *types.SystemContext) (types.ImageDestination, error) {
+	tempDirOciRef, err := createOciRef(ref.image)
```

Comment:
Throughout: For no _really good_ reason, it seems that the Go convention is to use all lowercase for an acronym at the start of an identifier, but all uppercase elsewhere. So `ociArchiveReference`, but `createOCIRef`.